java/org/contikios/cooja/Cooja.java

```diff
@@ -3612,7 +3612,7 @@ public void actionPerformed(ActionEvent e) {
      * @return Worker method return value
      */
     public T invokeAndWait() {
-      if(java.awt.EventQueue.isDispatchThread()) {
+      if (!Cooja.isVisualized() || java.awt.EventQueue.isDispatchThread()) {
```

Comment:
The purpose of `RunnableInEDT` is to run code on the AWT dispatch thread. Perhaps a better solution would be to throw an exception if Cooja is not visualized and then make sure `RunnableInEDT` is only used when Cooja is visualized? This would reduce the risk of GUI code being executed outside the AWT dispatch thread by mistake?

Reply:
I have updated this PR with the fix.

Will add the exception in a separate pull request after some more testing.